test/unit/app/controllers/permissions/permissions-controller-test.js

```diff
@@ -0,0 +1,1387 @@
+        [ORIGINS.a]: [PERM_NAMES.eth_accounts],
+        [ORIGINS.b]: [PERM_NAMES.eth_accounts],
+      })
+
+      aAccounts = await permController.getAccounts(ORIGINS.a)
+      bAccounts = await permController.getAccounts(ORIGINS.b)
+
+      assert.deepEqual(aAccounts, [], 'first origin should have no accounts')
+      assert.deepEqual(bAccounts, [], 'second origin should have no accounts')
+
+      assert.deepEqual(
+        notifications[ORIGINS.a], [NOTIFICATIONS.removedAccounts()],
+        'first origin should have correct notification'
+      )
+      assert.deepEqual(
+        notifications[ORIGINS.b], [NOTIFICATIONS.removedAccounts()],
+        'second origin should have correct notification'
+      )
+
+      assert.deepEqual(
+        Object.keys(permController.permissions.getDomains()), [],
+        'all domains should be deleted'
+      )
+    })
+
+    it('only removes targeted permissions from single domain', async function () {
+
+      grantPermissions(
+        permController, ORIGINS.b, PERMS.finalizedRequests.test_method()
+      )
+
+      let bPermissions = permController.permissions.getPermissionsForDomain(ORIGINS.b)
+
+      assert.ok(
+        (
+          bPermissions.length === 2 &&
+          find(bPermissions, { parentCapability: PERM_NAMES.eth_accounts }) &&
+          find(bPermissions, { parentCapability: PERM_NAMES.test_method })
+        ),
+        'origin should have correct permissions'
+      )
+
+      permController.removePermissionsFor({
+        [ORIGINS.b]: [PERM_NAMES.test_method],
+      })
+
+      bPermissions = permController.permissions.getPermissionsForDomain(ORIGINS.b)
+
+      assert.ok(
+        (
+          bPermissions.length === 1 &&
+          find(bPermissions, { parentCapability: PERM_NAMES.eth_accounts })
+        ),
+        'targeted permission only should have been removed'
+      )
+    })
+
+    it('removes permissions for a single domain, without affecting another', async function () {
+
+      permController.removePermissionsFor({
+        [ORIGINS.b]: [PERM_NAMES.eth_accounts],
+      })
+
+      const aAccounts = await permController.getAccounts(ORIGINS.a)
+      const bAccounts = await permController.getAccounts(ORIGINS.b)
+
+      assert.deepEqual(
+        aAccounts, ACCOUNT_ARRAYS.a,
+        'first origin does not have correct accounts'
+      )
+      assert.deepEqual(bAccounts, [], 'second origin should have no accounts')
+
+      assert.deepEqual(
+        notifications[ORIGINS.a], [],
+        'first origin should have no notifications'
+      )
+      assert.deepEqual(
+        notifications[ORIGINS.b], [NOTIFICATIONS.removedAccounts()],
+        'second origin should have correct notification'
+      )
+
+      assert.deepEqual(
+        Object.keys(permController.permissions.getDomains()), [ORIGINS.a],
+        'only first origin should remain'
+      )
+    })
+
+    // we do not check notifications in this test, because while the mocks will
```

Comment:
I rewrote this in a way that's hopefully more clear:
```
    // we do not check notifications in this test, because while our mock
    // notifications will be emitted, the ones used in production ignore
    // unknown domains
    // see getNotifyDomain and getNotifyAllDomains for details
```

That is to say, the `permController` instance isn't mocked, but the notification functions are.

Reply:
In the context of the permissions controller itself though, it does call `notifyDomain` - right? Whether or not `notifyDomain` does anything seems kinda... outside of scope for this test.

Reply:
Fixed in 565833a